CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+# Contributing to Snoopy
+
+This doc outlines the architecture of Snoopy and how provides a step-by-step
+guide of setting it up on your machine.
+
+## Setup
+
+Snoopy is a [Yarn workspace](https://yarnpkg.com/en/docs/workspaces) that is
+composed of several separate packages. These packages are all published to NPM
+together with [Lerna](https://github.com/lerna/lerna). First install all
+dependencies with:
+
+```
+yarn
+```
+
+To build Snoopy for development, you need to compile the codebase with
+[TypeScript](https://www.typescriptlang.org/). In the root of the project run:
+
+```
+yarn build:watch
+```
+
+This will watch and incrementally build all packages. If you only want to build
+without watching, drop the `:watch`.
+
+## Running on an app
+
+Snoopy itself is useless without an App to run it on. There are several examples
+included in the Snoopy codebase, but you can run it on any React project you
+want. As an example, we will use the `02-to-do-list` example.
+
+Navigate to the example:
+
+```
+cd examples/02-to-do-list
+```
+
+Start Snoopy by running node on the entry to the CLI package:
+
+```
+node ../../packages/cli/lib/index.js
+```
+
+You should see that the server has started on some port (default 3042). You can
+navigate to `localhost:PORT` and Snoopy should be running.
+
+## Making changes
+
+When you make a change to any package that is not the UI, make sure that the
+`build:watch` command has completed its incremental complication. Otherwise your
+changes will not be run. If there are any TypeScript issues and the command
+shows an error, your latest changes will not be run. Changes to the UI can be
+made while the server is still running and Parcel will update the UI with HMR.
+
+## Architecture
+
+Currently there are 7 packages apart of Snoopy. 5 of which are published to NPM.
+
+### api
+
+Contains names of all events that are sent between server and client (UI).
```

Comment:
It sounds counterintuitive to have an entire package just to share strings between other packages.  Why not just define those on the server repo and read from the client repo? Or alternatively, I could imagine having a "type" package to share a bunch of definitions across server and client, for instance `type SnoopyEvents = "open-file" | ... `

Reply:
The api package is used by the vscode plugin to send messages to the server. In this case it is basically a type package. However, if we eventually want to share more logic between server and plugin, it can go in the api package.